src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReceivedZipFileConverter.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.reports;
+
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.reports.ReceivedZipFile;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.ReceivedZipFileData;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.ZipFileIdentifier;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+
+public class ReceivedZipFileConverter {
+
+    public List<ReceivedZipFileData> convertReceivedZipFiles(List<ReceivedZipFile> receivedZipFiles) {
+        return receivedZipFiles.stream()
+            .collect(groupingBy(file ->
+                                    new ZipFileIdentifier(
+                                        file.getZipFileName(),
+                                        file.getContainer()
+                                    )
+            ))
```

Comment:
so this is grouping by new object all the time? doesn't it make 1-to-1 map here? perhaps we are better of with `new tuple2` here. moreover - we have vavr library - it certainly supports such grouping collection

Reply:
especially when it is dropped/disregarded next line - why even need extra model/class in our source code?